ecs-cli/modules/cli/cluster/cluster_app_test.go

```diff
@@ -191,6 +207,61 @@ func TestClusterUpWithoutPublicIP(t *testing.T) {
 	assert.NoError(t, err, "Unexpected error bringing up cluster")
 }
 
+func TestClusterUpWithUserData(t *testing.T) {
+	defer os.Clearenv()
```

Comment:
Do you know why we're calling this? I see this is called in existing tests, but it's not immediately apparent what's being set that we need to clear. And if there is something we want to clear, I'd prefer we unset that/those specifically instead of blowing away the whole ENV (example [here](https://github.com/allisaurus/amazon-ecs-cli/blob/regcreds/ecs-cli/modules/utils/regcreds/creds_input_reader_test.go#L82))

Reply:
It looks like all of the tests do it because the function [setupTest()](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/cli/cluster/cluster_app_test.go#L80) sets some environment variables.  We could specifically unset just those, but I do not see any problems with unsetting the entire environment at the end of the test. To ensure isolation, each test case invocation should start with an empty environment. 

Remember that `os.Clearnenv()` doesn't affect the env of the user in the shell, just the env for the running go program, which is a child process of the shell.

Reply:
Ok, thanks for clarifying (though how the [OS docs](https://golang.org/pkg/os/#Clearenv) are worded I was def worried about a larger scope of impact)